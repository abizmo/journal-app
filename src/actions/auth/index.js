export { default as login } from './login';
export { default as loginWithEmail } from './loginWithEmail';
export { default as loginWithGoogle } from './loginWithGoogle';
export { default as logout } from './logout';
export { default as signOut } from './signOut';
export { default as signUp } from './signUp';

export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export const types = {
  LOGIN,
  LOGOUT,
};
