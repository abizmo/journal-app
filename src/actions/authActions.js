import Swal from "sweetalert2";
import {
  loginWithGoogleProvider,
  signInWithEmail,
  signOutInFirebase,
  signUpWithEmailAndPassword
} from "../firebase";
import { startAction, stopAction } from "./uiActions";

export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export const login = (user) => ({
  type: LOGIN,
  payload: user,
});

export const logout = () => ({
  type: LOGOUT,
});

export const loginWithEmail = ({ email, password }) => (dispatch) => {
  dispatch(startAction());
  signInWithEmail({ email, password })
    .then((user) => {
      dispatch(stopAction());
      dispatch(login(user));
    })
    .catch(({ message }) => {
      dispatch(stopAction());
      Swal.fire('Error', message, 'error');
    });
};

export const loginWithGoogle = () => (dispatch) => {
  loginWithGoogleProvider()
    .then((user) => dispatch(login(user)))
    .catch(({ message }) => Swal.fire('Error', message, 'error'));
};

export const signUp = (newUser) => (dispatch) => {
  signUpWithEmailAndPassword(newUser)
    .then((user) => dispatch(login(user)))
    .catch(({ message }) => Swal.fire('Error', message, 'error'));
};

export const signOut = () => (dispatch) => {
  signOutInFirebase()
    .then(() => dispatch(logout()))
    .catch(({ message }) => Swal.fire('Error', message, 'error'));
};
