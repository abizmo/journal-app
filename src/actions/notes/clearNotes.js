import { CLEAR_NOTES } from ".";

const clearNotes = () => ({
  type: CLEAR_NOTES,
});

export default clearNotes;
