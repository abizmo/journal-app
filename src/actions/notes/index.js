export { default as setActive } from './setActive';
export { default as refreshNote } from './refreshNote';
export { default as clearNotes } from './clearNotes';
export { default as getNotes } from './getNotes';
export { default as uploadImage } from './uploadImage';
export { default as removeNote } from './removeNote';
export { default as newNote } from './newNote';
export { default as saveNote } from './saveNote';

export const ADD_NOTE = 'ADD_NOTE';
export const CLEAR_NOTES = 'CLEAR_NOTES'
export const DELETE_NOTE = 'DELETE_NOTE';
export const SET_ACTIVE = 'SET_ACTIVE';
export const SET_NOTES = 'SET_NOTES';
export const UPDATE_NOTE = 'UPDATE_NOTE';

export const types = {
  ADD_NOTE,
  CLEAR_NOTES,
  DELETE_NOTE,
  SET_ACTIVE,
  SET_NOTES,
  UPDATE_NOTE,
};
