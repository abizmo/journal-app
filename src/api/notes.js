import {
  addDoc,
  collection,
  doc,
  getDocs,
  updateDoc,
} from "firebase/firestore";
import { db } from "../firebase";

export const createNote = (userId, note) => addDoc(collection(db, `${userId}/journal/notes/`), note)
  .then(({ id }) => id);

export const getAllNotes = (userId) => getDocs(collection(db, `${userId}/journal/notes/`))
  .then((res) => {
    const notes = [];
    res.forEach((doc) => {
      notes.push({ ...doc.data(), id: doc.id });
    });
    
    return notes;
});

export const updateNote = (userId, noteId, data) => {
  const docRef = doc(db, `${userId}/journal/notes/${noteId}`);

  return updateDoc(docRef, data)
    .then(() => data);
};