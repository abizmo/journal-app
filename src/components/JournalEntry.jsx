import React from 'react'

const JournalEntry = () => {
  const imageUrl = 'https://images.unsplash.com/photo-1519699047748-de8e457a634e?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTZ8fHBlb3BsZXxlbnwwfDJ8MHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60';
  return (
    <div className="journal__entry-container">
      <div
        className="journal__entry-image"
        style={{ backgroundImage: `url(${imageUrl})`}}
      />
      <div className="journal__entry-body">
        <p className="journal__entry-title">
          Hola
        </p>
        <p className="journal__entry-content">
          Veniam sit veniam eiusmod velit aliqua qui officia ea aliquip adipisicing.
        </p>
      </div>
      <div className="journal__entry-daybox">
        <span className="journal__entry-text">Monday</span>
        <span className="journal__entry-number">28</span>
      </div>
    </div>
  )
}

export default JournalEntry
