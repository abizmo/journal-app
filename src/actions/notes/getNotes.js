import { SET_NOTES } from ".";
import { getAllNotes } from "../../api/notes";
import { error } from "../../helpers/alert";

const setNotes = (notes) => ({
  type: SET_NOTES,
  payload: notes,
});

const getNotes = () => (dispatch, getState) => {
  const { userId } = getState().auth;

  getAllNotes(userId)
    .then((notes) => {
      dispatch(setNotes(notes));
    })
    .catch(({ message }) => error(message));
};

export default getNotes;
