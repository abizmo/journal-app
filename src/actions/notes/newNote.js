import { ADD_NOTE } from ".";
import { createNote } from "../../api/notes";
import { error } from "../../helpers/alert";

const addNote = (note) => ({
  type: ADD_NOTE,
  payload: note,
});

const newNote = () => (dispatch, getState) => {
  const { userId } = getState().auth;

  const note = {
    title: '',
    body: '',
    date: new Date().getTime(),
  };

  createNote(userId, note)
    .then((id) => {
      dispatch(addNote({ ...note, id }))
    })
    .catch(({ message }) => error(message));
};

export default newNote;
