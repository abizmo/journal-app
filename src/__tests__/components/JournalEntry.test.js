import { mount } from "enzyme";
import { Provider } from "react-redux";
import createMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import { setActive } from "../../actions/notes";
import JournalEntry from "../../components/JournalEntry";
import { notes } from "../../fixtures/notes";
jest.mock('../../actions/notes');

const date = new Date().getTime();
const initialState = {};
const middlewares = [thunk];
const mockStore = createMockStore(middlewares);

describe('<JournalEntry />', () => {
  const store = mockStore(initialState);
  store.dispatch = jest.fn();

  const wrapper = mount(
    <Provider store={store}>
      <JournalEntry
        { ...notes[0] }
        date={ date }
      />
    </Provider>
  );

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.journal__entry-title').text().trim()).toBe(notes[0].title);
  });

  test('should dispatch setActive', () => {
    wrapper.find('.journal__entry-container').simulate('click');

    expect(setActive).toHaveBeenCalledWith({ ...notes[0], date });
  });
});
