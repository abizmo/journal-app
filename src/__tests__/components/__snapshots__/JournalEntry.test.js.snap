// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalEntry /> should render and match snapshot 1`] = `
<div
  className="journal__entry-container"
  onClick={[Function]}
>
  <div
    className="journal__entry-image"
    style={
      Object {
        "backgroundImage": "url(http://localhost/images/1.jpeg)",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Titulo 1
    </p>
    <p
      className="journal__entry-content"
    >
      Body 1
    </p>
  </div>
  <div
    className="journal__entry-daybox"
  >
    <span
      className="journal__entry-text"
    >
      Monday
    </span>
    <span
      className="journal__entry-number"
    >
      11th
    </span>
  </div>
</div>
`;
